app/src/main/java/org/wikipedia/gallery/GalleryActivity.kt

```diff
@@ -0,0 +1,761 @@
+            binding.licenseIcon.setImageResource(R.drawable.ic_license_cc)
+            binding.licenseIconBy.setImageResource(R.drawable.ic_license_by)
+            binding.licenseIconBy.visibility = View.VISIBLE
+            binding.licenseIconSa.setImageResource(R.drawable.ic_license_sharealike)
+            binding.licenseIconSa.visibility = View.VISIBLE
+        } else {
+            binding.licenseIcon.setImageResource(license.licenseIcon)
+            binding.licenseIconBy.visibility = View.GONE
+            binding.licenseIconSa.visibility = View.GONE
+        }
+
+        // Set the icon's content description to the UsageTerms property.
+        // (if UsageTerms is not present, then default to Fair Use)
+        binding.licenseIcon.contentDescription = StringUtils.defaultIfBlank(item.mediaInfo!!.metadata!!.licenseShortName(),
+            getString(R.string.gallery_fair_use_license))
+        // Give the license URL to the icon, to be received by the click handler (may be null).
+        binding.licenseIcon.tag = item.mediaInfo!!.metadata!!.licenseUrl()
+        DeviceUtil.setContextClickAsLongClick(binding.licenseContainer)
+        val creditStr = if (item.mediaInfo!!.metadata!!.artist().isNotEmpty()) item.mediaInfo!!.metadata!!.artist()
+        else item.mediaInfo!!.metadata!!.credit()
+
+        // if we couldn't find a attribution string, then default to unknown
+        binding.creditText.text = StringUtil.fromHtml(StringUtils.defaultIfBlank(creditStr, getString(R.string.gallery_uploader_unknown)))
+        binding.infoContainer.visibility = View.VISIBLE
+    }
+
+    private inner class GalleryItemAdapter(activity: AppCompatActivity?) : PositionAwareFragmentStateAdapter(activity!!) {
+        private val list: MutableList<MediaListItem> = ArrayList()
+        fun setList(list: List<MediaListItem>) {
+            this.list.clear()
+            this.list.addAll(list)
+            notifyDataSetChanged()
+        }
+
+        override fun getItemCount(): Int {
+            return list.size
+        }
+
+        override fun createFragment(position: Int): Fragment {
+            return GalleryItemFragment.newInstance(pageTitle, list[position])
+        }
+    }
+
+    private inner class MediaDownloadReceiverCallback : MediaDownloadReceiver.Callback {
+        override fun onSuccess() {
+            FeedbackUtil.showMessage(this@GalleryActivity, R.string.gallery_save_success)
+        }
+    }
+
+    companion object {
+        const val ACTIVITY_RESULT_PAGE_SELECTED = 1
+        private const val ACTIVITY_REQUEST_DESCRIPTION_EDIT = 2
+        const val ACTIVITY_RESULT_IMAGE_CAPTION_ADDED = 3
+        const val ACTIVITY_REQUEST_ADD_IMAGE_TAGS = 4
+        const val EXTRA_PAGETITLE = "pageTitle"
+        const val EXTRA_FILENAME = "filename"
+        const val EXTRA_WIKI = "wiki"
+        const val EXTRA_REVISION = "revision"
+        const val EXTRA_SOURCE = "source"
+        private var TRANSITION_INFO: ImageHitInfo? = null
```

Comment:
Please have these a proper order.

Reply:
the 1,2,3,4 wont be in order then.. is that ok?